<launch>


<!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find multi_robot_nav)/config/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="true"/>

  <arg name="first_tb3"  default="tb3_0"/>
  <arg name="second_tb3" default="tb3_1"/>

  <arg name="first_tb3_x_pos" default=" -1.2"/>
  <arg name="first_tb3_y_pos" default=" 0.0"/>
  <arg name="first_tb3_z_pos" default=" 0.0"/>
  <arg name="first_tb3_yaw"   default=" 0.0"/>

  <arg name="second_tb3_x_pos" default=" 1.2"/>
  <arg name="second_tb3_y_pos" default=" 0.0"/>
  <arg name="second_tb3_z_pos" default=" 0.0"/>
  <arg name="second_tb3_yaw"   default=" 0.0"/>
 
  <param name="/use_sim_time" value="true"/>

  <group ns = "$(arg first_tb3)">
   <param name="tf_prefix" value="$(arg first_tb3)"/>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)">
   <param name="frame_id" value="/map" />
  </node>

   <!-- AMCL -->
   <include file="$(find multi_robot_nav)/launch/localization_files/amcl_tb3_0.launch"/>

   <!-- Move base -->
   <include file="$(find multi_robot_nav)/launch/localization_files/move_base_tb3_0.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
   </include>

   <!-- Dynamic Action Server -->
   <node pkg="multi_robot_nav" name="dynamic_localization_server" type="dynamic_localization_server.py"/>
   <node pkg="multi_robot_nav" name="dynamic_localization_client" type="dynamic_localization_client.py"/>

   <!-- Static Action Server -->
   <node pkg="multi_robot_nav" name="static_localization_server" type="static_localization_server.py"/>

  </group>


  <group ns = "$(arg second_tb3)">
   <param name="tf_prefix" value="$(arg second_tb3)"/>

   <!-- Map server -->
   <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="/map" />
   </node>

   <!-- AMCL -->
   <include file="$(find multi_robot_nav)/launch/localization_files/amcl_tb3_1.launch"/>

   <!-- Move base -->
   <include file="$(find multi_robot_nav)/launch/localization_files/move_base_tb3_1.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
   </include>

   <!-- Dynamic Action Server -->
   <node pkg="multi_robot_nav" name="dynamic_localization_server" type="dynamic_localization_server.py"/>
   
   <!-- Static Action Server -->
   <node pkg="multi_robot_nav" name="static_localization_server" type="static_localization_server.py"/>

  </group>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find multi_robot_nav)/rviz/turtlebot3_navigation_multi.rviz"/>
  </group>

=
</launch>
